src/shared/hooks/usePermissions.tsx

```diff
@@ -22,7 +23,8 @@ export const usePermissions = (asyncFunc: () => Promise<any> | undefined) => {
 
         if (
           payload?.response?.status === ApiResponseCodes.Forbidden ||
-          payload?.status === ApiResponseCodes.Forbidden
+          payload?.status === ApiResponseCodes.Forbidden ||
+          payload?.[0]?.type === ErrorResponseType.AccessDenied
```

Comment:
if the payload can be a list, it's probably better to check an assumption in the array: `typeof payload === 'array' && payload.some(data => data.type === ErrorResponseType.AccessDenied)`